### PR TITLE
Use fake TK app over global state several tests

### PR DESCRIPTION
**Pull request by frenchy64**
**Epic** https://github.com/threatgrid/iroh/issues/3986

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: Use fake TK app over global state several tests
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: test/ctia/bundle/routes_test.clj [hunk 1]
```diff
@@ -908,17 +915,17 @@
...
+  (helpers/with-config-transformer*
+    #(-> %
+         (assoc-in [:ctia :access-control :min-tlp] tlp)
+         (assoc-in [:ctia :access-control :default-tlp] tlp))
+    f))
 
 (deftest bundle-tlp-test
-  (test-for-each-store
-   (fn []
+ (with-tlp-property-setting "amber"
```
Comment (frenchy64) [inline test/ctia/bundle/routes_test.clj:925]: We must bring `with-tlp-property-setting` outside `test-for-each-store` so the former can change the config via `with-config-transformer*`  before the latter starts an app.

File: test/ctia/test_helpers/core.clj [hunk 2]
```diff
@@ -194,7 +194,11 @@
 
 (defn build-get-in-config-fn []
   ;; stub implementation until trapperkeeper bootstrap
```
Comment (frenchy64) [inline test/ctia/test_helpers/core.clj:196]: A slightly more useful stub which is closer to the final impl.

File: test/ctia/bundle/routes_test.clj [hunk 3]
```diff
@@ -1002,4 +1009,4 @@
          (is (= 201 (:status incident-post-res)))
          (is (= 201 (:status link-res)))
          (is (= 200 (:status bundle-get-res)))
-         (is seq (-> bundle-get-res :parsed-body :casebooks)))))))
+         (is (seq (-> bundle-get-res :parsed-body :casebooks))))))))
```
Comment (vemv) [inline test/ctia/bundle/routes_test.clj:1012]: :( thanks for this one

I checked that there are no more occurences: `rg --pcre2 '\(is\s(?!\().{1}'`

File: test/ctia/entity/asset_test.clj [hunk 4]
```diff
@@ -22,8 +21,10 @@
 
 (defn additional-tests [asset-id asset-sample]
   (testing "GET /ctia/asset/search"
+   (let [app (helpers/get-current-app)
+         {:keys [get-in-config]} (helpers/get-service-map app :ConfigService)]
     ;; only when ES store
-    (when (= "es" (p/get-in-global-properties [:ctia :store :asset]))
+    (when (= "es" (get-in-config [:ctia :store :asset]))
```
Comment (vemv) [inline test/ctia/entity/asset_test.clj:27]: Maybe https://github.com/threatgrid/ctia/pull/965#discussion_r477188465

**Review by vemv — APPROVED** [review]
lgtm, btw :)

File: test/ctia/entity/asset_test.clj (hunk 4 above)
Comment (ereteog) [reply test/ctia/entity/asset_test.clj:27]: indeed there is no other store implemented yet, so we could simply replace all the patterns`(when (= "es" (get-in-config [:ctia :store store-kw])) body)` by  `body`.

**Review by ereteog — APPROVED** [review]
LGTM

File: test/ctia/entity/asset_test.clj (hunk 4 above)
Comment (ereteog) [reply test/ctia/entity/asset_test.clj:27]: And if we need to exclude a potential new store later we could use the `when-not` pattern. WDYT?

**Review by yogsototh — APPROVED** [review]
LGTM

File: test/ctia/entity/asset_test.clj (hunk 4 above)
Comment (vemv) [reply test/ctia/entity/asset_test.clj:27]: > And if we need to exclude a potential new store later we could use the when-not pattern. WDYT?

sure!

**Pull request merged**